frontend/src/metabase/query_builder/components/view/sidebars/QuestionInfoSidebar/components/QuestionSources.unit.spec.tsx

```diff
@@ -0,0 +1,145 @@
+import { Route } from "react-router";
+import _ from "underscore";
+
+import { mockSettings } from "__support__/settings";
+import { createMockEntitiesState } from "__support__/store";
+import { renderWithProviders, screen, within } from "__support__/ui";
+import { modelIconMap } from "metabase/lib/icon";
+import { checkNotNull } from "metabase/lib/types";
+import { getMetadata } from "metabase/selectors/metadata";
+import { convertSavedQuestionToVirtualTable } from "metabase-lib/v1/metadata/utils/saved-questions";
+import type { Card, NormalizedTable } from "metabase-types/api";
+import { createMockCard, createMockSettings } from "metabase-types/api/mocks";
+import { createSampleDatabase } from "metabase-types/api/mocks/presets";
+import { createMockState } from "metabase-types/store/mocks";
+
+import { QuestionSources } from "./QuestionSources";
+
+interface SetupOpts {
+  card?: Card;
+  sourceCard?: Card;
+}
+
+const setup = async ({
+  card = createMockCard(),
+  sourceCard,
+}: SetupOpts = {}) => {
+  const state = createMockState({
+    settings: mockSettings(createMockSettings()),
+    entities: createMockEntitiesState({
+      databases: [createSampleDatabase()],
+      questions: _.compact([card, sourceCard]),
+    }),
+  });
+
+  // 😫 all this is necessary to test a card as a question source
+  if (sourceCard) {
+    const virtualTable = convertSavedQuestionToVirtualTable(sourceCard);
```

Comment:
Just a nomenclature thing, it feels like this should be called a "virtual question", since it's a question that is really a table underneath, represented in metadata.tables.

Reply:
but the object is in the shape of a table. we're creating a fake table from a real question